### DISCUSSION TITLE
accessing yaml sequence metadta

### DISCUSSION THREAD

**Discussion by rputikar** [quarto-dev/quarto-cli#6111] [Q&A] [answered]
### Description

How can I access and iterate over the following metadata in a quarto document?

staff:
    - id: j1
      name: John Doe
    - id: j2
      name: Jane Doe

admin:
    staff:
        - ref: j1

testing:
    staff: 
        - ref: j1
        - ref: j2

I want to produce the following in the document: 

Admin Staff: John Doe.
Testing staff: John Doe, Jane Doe. 

Any pointers on the best way to implement this?

**Comment by cscheid** [comment, marked as answer]
There are at least two ways:

- you can define a new template, and then use [Pandoc's template syntax](https://pandoc.org/MANUAL.html#templates) to iterate over that list
- you can write a Lua filter that reads the values from the metadata and then inserts them in the proper place in the document.

Let me give you an example of the latter. In your document, you write:

```md
---
filters: 
 - filter.lua
admin:
  staff:
  - name: John Doe
  - name: Jane Doe
# note here that i'm not using the indirection you want; but you should
# be able to adapt this to your needs.
---

Admin Staff: []{#admin-placeholder}
```

```lua
-- filter.lua

local admin_staff
local capture_meta = {
  Meta = function(meta)
    admin_staff = meta.admin.staff
  end
}

local replace_list = {
  Span = function(span)
    if span.identifier == "admin-placeholder" then
      local result = pandoc.Inlines({})
      for i, v in ipairs(admin_staff) do
        result:insert(pandoc.Span(pandoc.utils.stringify(v.name)))
        if i < #admin_staff then
          result:insert(pandoc.Span(pandoc.Str(", ")))
        end
      end
      return result
    end
  end
}

return { capture_meta, replace_list }
```

This file produces this output:

<img width="291" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/814a0803-a2ac-4c69-8777-34bc302d84f6">

**Reply by rputikar** [reply]
Thanks a lot. much appreciated!
